### DISCUSSION TITLE
Questions: How often to receive / Server Push

### DISCUSSION THREAD

**Discussion by FAUSheppy** [AsamK/signal-cli#1274] [Q&A]
Is it possible to set this up for receiving pushes on new messages instead of only polling?
How often can I call receive? Is there a rate limit on calling receive?

i.e. would it be acceptable/possible to receive every 5s to react to incoming messages quickly?

**Comment by AsamK** [comment]
There's no rate limit on receive, as far as i know.

**Comment by neagix** [comment]
I think there are 2 efficient ways to receive messages:
1. run `signal-cli daemon`, then use it as a daemon
2. `signal-cli receive --max-messages 1 --timeout 60`

The second one is still operating in CLI style but it will return as soon as first message comes in.
